services/src/main/scala/cromwell/pipeline/service/GitLabProjectVersioning.scala

```diff
@@ -14,13 +16,41 @@ class GitLabProjectVersioning(httpClient: HttpClient, config: GitLabConfig)
 
   override def updateFile(project: Project, projectFile: ProjectFile)(
     implicit ec: ExecutionContext
-  ): AsyncResult[String] = ???
+  ): AsyncResult[String] = {
+    val path = URLEncoder.encode(projectFile.path.toString, "UTF-8")
+    val fileUrl = s"${config.url}projects/${project.repository}/repository/files/$path"
+
+    httpClient
+      .put(
+        fileUrl,
+        payload = Json.stringify(Json.toJson(UpdateFileRequest("master", projectFile.content))),
```

Comment:
Did you discuss with @IaroslavTavchenkov what will you do with the hard-coded `"master"` branch?

Reply:
Oh, I thought we should discuss how to parse payload to json. Will discuss then